### PR TITLE
Update linter rules

### PR DESCRIPTION
**Pull request by fbernaly**
Update linter rules to make code more production ready and conform with Flutter standards.

### CODE REVIEW

**Comment by fbernaly** [comment]
@obrunsmann: these changes are just to polish the code, I did not change anything with the functionality or the logic in the plugin. For example I remove the print and change then for `_log.info` to allow plugin consumer to turn on/off the logs if needed.

**Comment by obrunsmann** [comment]
@fbernaly Thanks for your contribution! LGTM, give me just another 1-2 days to double check the huge amount of additions and then I'll merge 💪 But these styles fit in any case with what I otherwise also use

**Comment by fbernaly** [comment]
@obrunsmann : sure,  it was just formatting and enforcing linter rules, no change to the code. Once this one is merge I will send a PR for the chat feature (still in progress).

File: analysis_options.yaml [hunk 1]
```diff
@@ -1 +1,14 @@
-include: package:pedantic/analysis_options.yaml
+include: package:flutter_lints/flutter.yaml
+
+analyzer:
+  exclude: [ lib/src/api/proto/**/*.dart,
+             lib/**/*.swagger*dart]
+
+linter:
+  rules:
+    prefer_relative_imports: true
```
Comment (obrunsmann) [inline analysis_options.yaml:9]: Personally I am not a fan of using relative paths which is also not part of the new `flutter_lints` package requirements. 

Relative parts in small projects are okay, but when growing we experienced imports like:

```dart
import '../../../../../../widgets/foo.dart';
```

Which is impossible to scan at least for my eyes :D

File: analysis_options.yaml [hunk 2]
```diff
@@ -1 +1,14 @@
-include: package:pedantic/analysis_options.yaml
+include: package:flutter_lints/flutter.yaml
+
+analyzer:
+  exclude: [ lib/src/api/proto/**/*.dart,
```
Comment (obrunsmann) [inline analysis_options.yaml:4]: Suggest using yml list
```yaml
analyzer:
  exclude:
      - lib/src/api/proto/**/*.dart
      -  ...
```

**Review by obrunsmann — CHANGES_REQUESTED** [review]
Just 2 small ideas here, apart from that looks good 👍

File: analysis_options.yaml (hunk 2 above)
Comment (fbernaly) [reply analysis_options.yaml]: addressed in https://github.com/obrunsmann/flutter_nakama/pull/26/commits/efd1d3bfba26e403efc2c3236702a3b06b4ab635

**Review by obrunsmann — APPROVED** [review]

**Comment by fbernaly** [comment]
@obrunsmann : thanks

**Pull request merged**